tests/Common/BranchAwareClientTest.php

```diff
@@ -86,7 +116,18 @@ public function testInvalidBranch()
             $options['url']
         );
         $this->expectException(InvalidArgumentException::class);
-        $this->expectExceptionMessage('Branch "" is not valid.');
-        new BranchAwareClient('', $options);
+        $this->expectExceptionMessage(sprintf('Branch "%s" is not valid.', $branchId));
+        new BranchAwareClient($branchId, $options);
+    }
+
+    /**
+     * @return array
+     */
+    public function provideInvalidBranchId()
+    {
+        return [
+            'empty string' => [''],
+            'empty number' => [0],
```

Comment:
co je to empty number? proč ne `invalid id`
přidal bych ještě case `'invalid placeholder' => ['main']` ať je jasné, že tam nejde dat jakýkoliv string

Reply:
Nepridal, protoze se v constructoru kontroluje jen `empty()`, nic vic a tam jsem nesahal...

Reply:
jo vlastně ten test není integrační, ale unit :+1: